src/pubsub/ua_pubsub_reader.c

```diff
@@ -778,7 +801,19 @@ DataSetReader_processRaw(UA_Server *server, UA_ReaderGroup *rg,
         const UA_DataType *type =
             UA_findDataTypeWithCustom(&dsr->config.dataSetMetaData.fields[i].dataType,
                                       server->config.customDataTypes);
-        msg->data.keyFrameData.rawFields.length += type->memSize;
+        size_t fieldLength = 0;
+        if((type->typeKind == UA_DATATYPEKIND_STRING || type->typeKind == UA_DATATYPEKIND_BYTESTRING) 
+            && (dsr->config.dataSetMetaData.fields[i].maxStringLength != 0))
+        {
+            // Length of binary encoded string = 4 (string length encoded as uint32) + N (actual string data) bytes.
```

Comment:
Add an assertion here that the max string length is always set for strings.
Because we do now check this in the config.